server/data/data-library/src/main/java/org/apache/james/user/lib/model/Algorithm.java

```diff
@@ -27,63 +27,53 @@
 public class Algorithm {
 
     public enum HashingMode {
-        LEGACY(LEGACY_FACTORY),
-        DEFAULT(DEFAULT_FACTORY);
+        PLAIN,
+        SALTED,
+        LEGACY,
+        LEGACY_SALTED;
```

Comment:
LEGACY was more kept for like not breaking old implementations but I don't think we want to rely on it for implementing something new like salted passwords? 

Can you confirm @chibenwa ?

Reply:
I believe there are a lot of people like me out there stuck with a LEGACY password database, since there was no way of getting rid of it before. This patch can help, but must be able to handle the LEGACY modes as the starting point before moving on to PLAIN or preferably SALTED.

Reply:
Ah no no I mean should keep LEGACY, I'm more dubious on why introducing a LEGACY_SALTED ? If people are on legacy and want to move, would be better to have them moving as you said on PLAIN or SALTED, but I don't see why should move towards a LEGACY_SALTED though?

Reply:
LEGACY_SALTED is what we currently have in our password database, since we added salting before we knew that it also was legacy mode. It is highly unlikely anybody else will wind up with this, but I saw no harm including the option anyway. If you believe this might be too confusing, I can remove LEGACY_SALTED and keep it as a private patch, until all our users finally updated their passwords. Just let me know.

Reply:
👍 let's keep legacy salted as it help the original contributor ;-) this topic was btw alredy discussed on the mailing list.

Reply:
OK, I'll keep it in then :-)

Reply:
Sorry then, I initially thought it was not used, but if you do it's alright ;)